src/Symfony/Component/Security/Core/User/User.php

```diff
@@ -117,4 +117,44 @@ public function isEnabled()
     public function eraseCredentials()
     {
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isEqualTo(UserInterface $user)
+    {
+        if (!$user instanceof self) {
+            return false;
+        }
+
+        if ($this->getPassword() !== $user->getPassword()) {
+            return false;
+        }
+
+        if ($this->getSalt() !== $user->getSalt()) {
+            return false;
+        }
+
+        if ($this->getUsername() !== $user->getUsername()) {
+            return false;
+        }
+
+        if ($this->isAccountNonExpired() !== $user->isAccountNonExpired()) {
```

Comment:
We can't assume that this method exists on `$user` class since it can be any `UserInterface` implementation